### PR TITLE
Added the deploy and undeploy commands

### PR DESCRIPTION
**Pull request by 1akshitha**
## Purpose
Added the support to deploy and undeploy revisions through ctl. This is achieved with the introduction of two commands, deploy and undeploy.

```apictl deploy api -e dev -n PizzaAPI -v 1.0.0 --rev 1```
```apictl undeploy api -e dev -n PizzaAPI -v 1.0.0 --rev 1```

Related issue: #601

### CODE REVIEW

File: import-export-cli/cmd/deployAPI.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+// DeployAPI command related usage info
+const DeployAPICmdLiteral = "api"
+const deployAPICmdShortDesc = "Deploy API"
+
+const deployAPICmdLongDesc = "Deploy an API to the given gateway environment"
+
+const deployAPICmdExamples = utils.ProjectName + ` ` + DeployCmdLiteral + ` ` + DeployAPICmdLiteral + ` -n TwitterAPI -v 1.0.0 -r admin --rev 1 -g Label1 -e dev
+` + utils.ProjectName + ` ` + DeployCmdLiteral + ` ` + DeployAPICmdLiteral + ` -n FacebookAPI -v 2.1.0 --rev 6 -g Label1 Label2 Label3 -e production
+` + utils.ProjectName + ` ` + DeployCmdLiteral + ` ` + DeployAPICmdLiteral + ` -n FacebookAPI -v 2.1.0 --rev 2 -r admin -g Label1 -e production --hide-on-devportal
+NOTE: All the 5 flags (--name (-n), --version (-v) , --rev and --gateway (-g) --environment (-e)) are mandatory.`
```
Comment (wasuradananjith) [inline import-export-cli/cmd/deployAPI.go:49]:
```suggestion
NOTE: All the 5 flags (--name (-n), --version (-v) , --rev, --gateway (-g) and --environment (-e)) are mandatory.`
```

Comment (wasuradananjith) [reply import-export-cli/cmd/deployAPI.go:49]: Shall we change like this?

File: import-export-cli/cmd/deployAPI.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	DeployAPICmd.Flags().StringVarP(&deployCmdAPIVersion, "version", "v", "",
+		"Version of the API to be deployed")
+	DeployAPICmd.Flags().StringVarP(&deployCmdProvider, "provider", "r", "",
+		"Provider of the API")
+	DeployAPICmd.Flags().StringVarP(&deployCmdGateway, "gateway", "g", "",
+		"Gateways which the revision has to be deployed")
+	DeployAPICmd.Flags().StringVarP(&deployCmdRevisionNum, "rev", "", "",
+		"Revision number of the API to be deployed")
+	DeployAPICmd.Flags().BoolVar(&deployCmdHideOnDevportal, "hide-on-devportal", false,
+		"Hide the gateway environment on devportal")
```
Comment (wasuradananjith) [inline import-export-cli/cmd/deployAPI.go:105]:
```suggestion
		"Hide the gateway environment on the Developer Portal")
```

Comment (wasuradananjith) [reply import-export-cli/cmd/deployAPI.go:105]: Shall we change devportal to Developer Portal? It is better if we can use the full term (Developer Portal) in the explanations and documentation as recommended by the docs team.

File: import-export-cli/cmd/undeployAPI.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+// UndeployAPICmd command related usage info
+const UndeployAPICmdLiteral = "api"
+const undeployAPICmdShortDesc = "Undeploy API"
+
+const undeployAPICmdLongDesc = "Undeploy an API to the given gateway environment"
+
+const undeployAPICmdExamples = utils.ProjectName + ` ` + UndeployCmdLiteral + ` ` + UndeployAPICmdLiteral + ` -n TwitterAPI -v 1.0.0 -r admin -g Label1 -e dev
+` + utils.ProjectName + ` ` + UndeployCmdLiteral + ` ` + UndeployAPICmdLiteral + ` -n FacebookAPI -v 2.1.0 --rev 6 --all-gateways -e production
+` + utils.ProjectName + ` ` + UndeployCmdLiteral + ` ` + UndeployAPICmdLiteral + ` -n FacebookAPI -v 2.1.0 --rev 2 -g Label1 -e production
+NOTE: All the 4 flags (--name (-n), --version (-v), --rev, --environment (-e) and one from --gateway (-g) or --all-gateways) are mandatory.`
```
Comment (wasuradananjith) [inline import-export-cli/cmd/undeployAPI.go:49]:
```suggestion
NOTE: All the 4 flags (--name (-n), --version (-v), --rev, --environment (-e)) and one from (--gateway (-g) or --all-gateways) are mandatory.`
```

Comment (wasuradananjith) [reply import-export-cli/cmd/undeployAPI.go:49]: Shall we change like this?

File: import-export-cli/cmd/deployAPI.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+		utils.Logf(utils.LogPrefixInfo+"ResponseStatus: %v\n", resp.Status())
+		if resp.StatusCode() == http.StatusCreated {
+			fmt.Println(apiNameForStateChange + " API state changed successfully!")
+		} else {
+			fmt.Println("Error while deploying the API: ", resp.Status(), "\n", string(resp.Body()))
+		}
+	} else {
+		fmt.Println("Error getting OAuth tokens to deploy the API:" + preCommandErr.Error())
+	}
+
```
Comment (Chamindu36) [inline import-export-cli/cmd/deployAPI.go:88]: shall we remove this extra line?

File: import-export-cli/cmd/undeployAPI.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+		utils.Logf(utils.LogPrefixInfo+"ResponseStatus: %v\n", resp.Status())
+		if resp.StatusCode() == http.StatusCreated {
+			fmt.Println(apiNameForStateChange + " API artifact successfully undeployed from the gateway!")
+		} else {
+			fmt.Println("Error while undeploying the  API: ", resp.Status(), "\n", string(resp.Body()))
+		}
+	} else {
+		fmt.Println("Error getting OAuth tokens to undeploy the API:" + preCommandErr.Error())
+	}
+
```
Comment (Chamindu36) [inline import-export-cli/cmd/undeployAPI.go:95]: Shall we remove this extra line?

**Comment by 1akshitha** [comment]
Closing this PR and providing the functionality through https://github.com/wso2/product-apim-tooling/pull/609/files

**Pull request closed without merging**